RachelAbaniwo/More-Recipes
# #153940836 refactor-client-side-code

#### What does this PR do?
refactor client side code

#### Description of Task to be completed?
- connect each component to only neccesary props
- change some features of create-recipe template
- change validation error messages for sign up, sign in and create recipe components
- move client/assets/js/script.js file content to client/index.html  file

#### How should this be manually tested?
* from terminal, run: `git clone https://github.com/RachelAbaniwo/More-Recipes.git` to clone 
   this repository
* cd into the cloned repository and run: `npm install` to install all required dependencies
* create a `.env` file in the root directory and set up postgres configurations as described in the `.env.sample` file.
* run `npm run start:dev` to start the server
* run `npm run start:client` to start the application
* navigate to localhost:8080/siginin (or any path specified in the client/src/app.js file) 
* inspect the screen using react dev tools, check the props, each screen has only the props it 
 needs.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#153940836

#### Screenshots (if appropriate)
- before refactoring
<img width="1440" alt="screen shot 2017-12-28 at 8 46 10 am" src="https://user-images.githubusercontent.com/32877365/34414997-9a7404ee-ebec-11e7-9ee3-7eb6233b4d82.png">


-after refactoring
<img width="1435" alt="screen shot 2017-12-28 at 4 29 44 pm" src="https://user-images.githubusercontent.com/32877365/34414972-7c9cfd2c-ebec-11e7-806d-5110b9b7a52f.png">


#### Questions:
N/A
